### ISSUE TITLE
Error on  RNNLearner.language_model(...)

### ISSUE THREAD

**Issue by ohmeow** [fastai/fastai#939]
I'm using the "TextLMDataBunch.from_df()" method ... and while I can iterate through the mini-batches and they all look good, I'm not getting an error when I run the following.  Everything was working fine until I got the latest.

```
learn = RNNLearner.language_model(data_lm, pretrained_fnames=['lstm_wt103', 'itos_wt103'], drop_mult=0.5)
learn.fit_one_cycle(1, 1e-2)
```
... was working fine before but is now throwing an exception ...

"Target size (torch.Size([5920])) must be the same as input size (torch.Size([5920, 25951]))"

<img width="1020" alt="screen shot 2018-10-22 at 11 04 10 pm" src="https://user-images.githubusercontent.com/14000/47339055-e1ddc200-d64e-11e8-8428-51d1fabeed4e.png">

**Comment by ohmeow** [comment]
Just noticed something weird ...

When I run "learn.loss_fn" it returns:

<function torch.nn.functional.binary_cross_entropy_with_logits(input, target, weight=None, size_average=None, reduce=None, reduction='elementwise_mean', pos_weight=None)>

That is unexpected as I'm not setting this anywhere and was expecting the default loss function of cross entropy for the LM

**Comment by brtknr** [comment]
Whats changed between when it was working fine and throwing error?

**Comment by sgugger** [comment]
There might have been some issue with the way TextDataSet decides is loss function. I pushed a fix, but I'm not sure it's going to work. Please reopen if it doesn't!
